src/ucp/rndv/rndv.c

```diff
@@ -333,15 +329,17 @@ static void ucp_rndv_complete_rma_put_zcopy(ucp_request_t *sreq, int is_frag_put
     } else {
         ucp_rkey_destroy(sreq->send.rndv.rkey);
 
-        atp_req = ucp_request_get(sreq->send.ep->worker);
-        if (ucs_unlikely(atp_req == NULL)) {
-            ucs_fatal("failed to allocate request for sending ATP");
-        }
+        if (ucs_likely(status == UCS_OK)) {
+            atp_req = ucp_request_get(sreq->send.ep->worker);
+            if (ucs_unlikely(atp_req == NULL)) {
+                ucs_fatal("failed to allocate request for sending ATP");
+            }
 
-        atp_req->send.ep = sreq->send.ep;
-        atp_req->flags   = 0;
-        ucp_rndv_req_send_ack(atp_req, sreq, sreq->send.rndv.remote_req_id,
-                              status, UCP_AM_ID_RNDV_ATP, "send_atp");
+            atp_req->send.ep = sreq->send.ep;
+            atp_req->flags   = 0;
+            ucp_rndv_req_send_ack(atp_req, sreq, sreq->send.rndv.remote_req_id,
+                                  status, UCP_AM_ID_RNDV_ATP, "send_atp");
+        }
```

Comment:
why not sending ATP anyway?

Reply:
I've aligned behavior with RNDV GET Zcopy which doesn't send ATS back in case of error.

Reply:
why not vice versa?

Reply:
need to revert non-essential change

Reply:
sorry forgot about it
done